d2graph/d2graph.go

```diff
@@ -1582,3 +1582,316 @@ func (g *Graph) SortEdgesByAST() {
 	})
 	g.Edges = edges
 }
+
+func (g *Graph) Compare(other *Graph) error {
```

Comment:
I think these should go in `serde`, everywhere else I wouldn't compare this way, I'd just do equality check or compare IDs.